plaidml/edsl/ffi.cc

```diff
@@ -973,7 +976,16 @@ plaidml_program* plaidml_program_evaluate(  //
       }
       mutations.updates.emplace(ProgramUpdate{src_updates[i]->value, dst_updates[i]->value});
     }
-    auto ret = new plaidml_program{GlobalContext::get()->MakeProgram(name, mutations)};
+
+    if (!pmlc::util::isFloat(static_cast<pmlc::util::DataType>(floatx))) {
```

Comment:
I think we should try to `symbolizeDataType` just before this line, so then we can call `isFloat()` and then pass it (now already converted to `pmlc::util::DataType` to the call to `MakeProgram()`